snapcraft/tests/test_project_loader.py

```diff
@@ -1838,10 +1838,44 @@ def test_valid_app_daemons(self):
                 'command': 'binary5',
                 'daemon': 'notify',
             },
+            'service6': {
+                'command': 'binary6',
+                'post-stop-command': 'binary6 --post-stop',
+                'daemon': 'simple'
+            },
         }
 
         project_loader.Validator(self.data).validate()
 
+    def test_daemon_dependency(self):
+        self.data['apps'] = {
+            'service1': {
+                'command': 'binary1',
+                'stop-command': 'binary1 --stop',
+            },
+        }
+        raised = self.assertRaises(
+            errors.SnapcraftSchemaError,
+            project_loader.Validator(self.data).validate)
+        self.assertEqual(
+            "The 'apps/service1' property does not match the required schema: "
+            "'daemon' is a dependency of 'stop-command'",
+            str(raised))
+
+        self.data['apps'] = {
+            'service1': {
+                'command': 'binary1',
+                'post-stop-command': 'binary1 --post-stop',
+            },
```

Comment:
@elespike thank you! It would be better if you use test scenarios here.

basically, you make a separate test case class for these two tests and you define a class variable called scenarios, like:

```
scenarios = [
    ('stop-command', {apps: {
        'service1': {...}
    )
    ('post-stop-command', {apps: {
        'service1': {...}
    )
]
```

Search for the word `scenarios` in other test files to get an idea of how we use them, and I'm available in https://rocket.ubuntu.com/channel/snapcraft if you need a hand.